Metaphlan
--
For more details of the tool: https://github.com/biobakery/biobakery/wiki/metaphlan3

**NOTE** this workflow only accepts single input files for now!

## Input files 
* MetaPhlAn accepts metagenomic sequence in several formats, including .fasta, .fastq, bowtie2out and sam.

## Quickstart
### Using MetaPhlAn on LatchBio
1. Go to https://latch.bio, create an account or log in
2. Search for `MetaPhlAn` via the `Explore` tab. See here https://console.latch.bio/explore/64842/info
3. Add `MetaPhlAn` to your workspace
4. Go to the `Workflows` tab, click on `MetaPhlAn`
5. Select the files you want to input into the workflow and select the number of processors to be used.
6. Click on `Launch Workflow`.


## Links
- LatchBio: https://latch.bio
- Source code: https://github.com/kriitoh/metaphlan

